bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/communication/RotelFlagsMapping.java

```diff
@@ -23,174 +27,80 @@
-     */
-    public boolean isMultiInputOn(byte[] flags) throws RotelException {
-        return RotelFlagsMapping.isBitFlagOn(flags, multiInputFlagNumber, multiInputBitNumber);
-    }
-
-    /**
-     * Set the multi input indicator
-     *
-     * @param flags the table of flags
-     * @param on true to set the indicator to ON or false to set it to OFF
-     *
-     * @throws RotelException in case the multi input indicator is undefined
-     */
-    public void setMultiInput(byte[] flags, boolean on) throws RotelException {
-        RotelFlagsMapping.setBitFlag(flags, multiInputFlagNumber, multiInputBitNumber, on);
-    }
-
-    /**
-     * Get the zone 2 indicator
-     *
-     * @param flags the table of flags
-     *
-     * @return true if the indicator is ON in the flags or false if OFF
-     *
-     * @throws RotelException in case the zone 2 indicator is undefined
-     */
-    public boolean isZone2On(byte[] flags) throws RotelException {
-        return RotelFlagsMapping.isBitFlagOn(flags, zone2FlagNumber, zone2BitNumber);
-    }
-
-    /**
-     * Set the zone 2 indicator
-     *
-     * @param flags the table of flags
-     * @param on true to set the indicator to ON or false to set it to OFF
-     *
-     * @throws RotelException in case the zone 2 indicator is undefined
-     */
-    public void setZone2(byte[] flags, boolean on) throws RotelException {
-        RotelFlagsMapping.setBitFlag(flags, zone2FlagNumber, zone2BitNumber, on);
+    public static final RotelFlagsMapping MAPPING1 = new RotelFlagsMapping(
+            List.of(new RotelFlagInfo(RotelFlagInfoType.MULTI_INPUT, 3, 1), //
+                    new RotelFlagInfo(RotelFlagInfoType.ZONE2, 5, 0), //
+                    new RotelFlagInfo(RotelFlagInfoType.CENTER, 8, 6), //
+                    new RotelFlagInfo(RotelFlagInfoType.SURROUND_LEFT, 8, 4), //
+                    new RotelFlagInfo(RotelFlagInfoType.SURROUND_RIGHT, 8, 3)));
+    public static final RotelFlagsMapping MAPPING2 = new RotelFlagsMapping(
+            List.of(new RotelFlagInfo(RotelFlagInfoType.ZONE2, 4, 7), //
+                    new RotelFlagInfo(RotelFlagInfoType.CENTER, 5, 6), //
+                    new RotelFlagInfo(RotelFlagInfoType.SURROUND_LEFT, 5, 4), //
+                    new RotelFlagInfo(RotelFlagInfoType.SURROUND_RIGHT, 5, 3)));
+    public static final RotelFlagsMapping MAPPING3 = new RotelFlagsMapping(
+            List.of(new RotelFlagInfo(RotelFlagInfoType.ZONE2, 3, 2), //
+                    new RotelFlagInfo(RotelFlagInfoType.ZONE3, 4, 2), //
+                    new RotelFlagInfo(RotelFlagInfoType.ZONE4, 4, 1), //
+                    new RotelFlagInfo(RotelFlagInfoType.CENTER, 5, 6), //
+                    new RotelFlagInfo(RotelFlagInfoType.SURROUND_LEFT, 5, 4), //
+                    new RotelFlagInfo(RotelFlagInfoType.SURROUND_RIGHT, 5, 3)));
+
+    private Map<RotelFlagInfoType, RotelFlagInfo> infos;
```

Comment:
```suggestion
    private Map<RotelFlagInfoType, RotelFlagInfo> infoMap;
```

(or similar, since plural of info is info)